extension/webviews/process-editor/src/ivy-diagram-widget.ts

```diff
@@ -36,15 +39,33 @@ export abstract class IvyGLSPDiagramWidget extends GLSPDiagramWidget {
     this.actionDispatcher.dispatch(EnableToolPaletteAction.create());
     this.actionDispatcher.dispatch(EnableViewportAction.create());
 
-    this.messenger.onRequest(InitializeServerRequest, server => this.initServer(server));
+    this.messenger.onRequest(InitializeConnectionRequest, () => this.initConnection());
     this.messenger.sendNotification(WebviewReadyNotification, HOST_EXTENSION);
   }
 
-  private initServer(server: string) {
+  private initConnection() {
     if (this.actionDispatcher instanceof GLSPActionDispatcher) {
-      this.actionDispatcher
-        .onceModelInitialized()
-        .finally(() => this.actionDispatcher.dispatch(EnableInscriptionAction.create({ connection: { server } })));
+      this.actionDispatcher.onceModelInitialized().finally(() => {
+        const ivyScript = {
+          reader: new WebSocketMessageReader(this.wrapWebSocket(IvyScriptWebSocketMessage)),
+          writer: new WebSocketMessageWriter(this.wrapWebSocket(IvyScriptWebSocketMessage))
+        };
+        const inscription = {
+          reader: new WebSocketMessageReader(this.wrapWebSocket(InscriptionWebSocketMessage)),
+          writer: new WebSocketMessageWriter(this.wrapWebSocket(InscriptionWebSocketMessage))
+        };
+        this.actionDispatcher.dispatch(EnableInscriptionAction.create({ connection: { ivyScript, inscription } }));
+      });
     }
   }
+
+  private wrapWebSocket(notificationType: NotificationType<string>): IWebSocket {
```

Comment:
I think it might be a bit cleaner to actually write our own `MessageReader` and `MessageWriter` class instead of mimicking a WebSocket that isn't one but really it is only the naming that makes it confusing so I can understand if we keep it like this. Again, I'm not entirely sure I see who is using the resulting MessageConnection.

Reply:
Good point, I will provide a MessageReader and Writer.